resources/net.go

```diff
@@ -0,0 +1,726 @@
+		checkOK = false
+	}
+
+	// check the ip addresses
+	if c, err := obj.ipCheckApply(apply); err != nil {
+		return false, err
+	} else if !c {
+		checkOK = false
+	}
+
+	return checkOK, nil
+}
+
+// NetUID is a unique resource identifier.
+type NetUID struct {
+	// NOTE: There is also a name variable in the BaseUID struct, this is
+	// information about where this UID came from, and is unrelated to the
+	// information about the resource we're matching. That data which is
+	// used in the IFF function, is what you see in the struct fields here.
+	BaseUID
+	name string // the network interface name
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *NetUID) IFF(uid ResUID) bool {
+	res, ok := uid.(*NetUID)
+	if !ok {
+		return false
+	}
+	return obj.name == res.name
+}
+
+// UIDs includes all params to make a unique identification of this object.
+// Most resources only return one although some resources can return multiple.
+func (obj *NetRes) UIDs() []ResUID {
+	x := &NetUID{
+		BaseUID: BaseUID{Name: obj.GetName(), Kind: obj.GetKind()},
+		name:    obj.Name,
+	}
+	return []ResUID{x}
+}
+
+// GroupCmp returns whether two resources can be grouped together or not.
+func (obj *NetRes) GroupCmp(r Res) bool {
+	_, ok := r.(*NetRes)
+	if !ok {
+		return false
+	}
+	return false
+}
+
+// Compare two resources and return if they are equivalent.
+func (obj *NetRes) Compare(r Res) bool {
+	// we can only compare NetRes to others of the same resource kind
+	res, ok := r.(*NetRes)
+	if !ok {
+		return false
+	}
+	if !obj.BaseRes.Compare(res) {
+		return false
+	}
+	if obj.Name != res.Name {
+		return false
+	}
+	if obj.State != res.State {
+		return false
+	}
+	if obj.IPv4 != res.IPv4 {
+		return false
+	}
+	if obj.IPv6 != res.IPv6 {
+		return false
+	}
+	if obj.Gateway != res.Gateway {
+		return false
+	}
+	if obj.Timeout != res.Timeout {
+		return false
+	}
+
+	return true
+}
+
+// UnmarshalYAML is the custom unmarshal handler for this struct.
+// It is primarily useful for setting the defaults.
+func (obj *NetRes) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawRes NetRes // indirection to avoid infinite recursion
+
+	def := obj.Default()     // get the default
+	res, ok := def.(*NetRes) // put in the right format
+	if !ok {
+		return fmt.Errorf("could not convert to NetRes")
+	}
+	raw := rawRes(*res) // convert; the defaults go here
+
+	if err := unmarshal(&raw); err != nil {
+		return err
+	}
+
+	*obj = NetRes(raw) // restore from indirection with type conversion!
+	return nil
+}
+
+// Close cleans up some resources so we can exit cleanly.
+func (obj *NetRes) Close() error {
+	// close the dbus connection
+	obj.sdbus.Close()
+	// close the message bus
+	obj.gdbus.Close()
+
+	return obj.BaseRes.Close()
+}
+
+// unitFileContents builds the unit file contents from the input
+func unitFileContents(name, ipv4, ipv6, gateway string) []byte {
+	// build the unit file contents
+	u := "[Match]\n"
```

Comment:
Instead of adding the `\n` everywhere, how about we build up a list and join with \n?